File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys
max = int(sys.stdin.readline())
stack=[]
result=[]
cnt=1

ans=0
for i in range(max):
    num=sys.stdin.readline().split()
    while cnt<=max or len(stack)>0:
        if cnt!=int(num[0])and cnt<=int(num[0])and cnt<=max:
            stack.append(cnt)
            result.append("+")
            cnt+=1
        elif cnt==int(num[0]):
            result.append("+")
            result.append("-")
            cnt+=1
            break
        elif len(stack)>0 and stack[len(stack)-1]== int(num[0]) :
            result.append("-")
            stack.pop()
            break
        else:
            ans=1
            break
if ans==0:
    for i in result:
        print(i)          
else:
    print("NO")

File: 백준/Silver/1874. 스택 수열/스택 수열.java

    import java.io.IOException;
    import java.io.BufferedReader;
    import java.io.InputStreamReader;
    import java.util.ArrayList;
    import java.util.LinkedList;
    import java.util.List;

    public class Main {
        public static void main(String[] args) throws IOException {

            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
            int len = Integer.parseInt(reader.readLine());
            LinkedList<Integer> stack = new LinkedList<>();
            ArrayList<String> result = new ArrayList<>();
            int current=1;
            for (int i=0; i <len; i++) {
                int command =Integer.parseInt(reader.readLine());

                while (current<=command){
                        stack.push(current);
                        result.add("+");
                        current++;
                    }
                if (stack.peek()==command) {
                        result.add("-");
                        stack.pop();
                }else{
                    System.out.println("NO");
                    return;
                }
            }
            for (String operation : result) {
                System.out.println(operation);
            }
            }
        }


